### PR TITLE
<TBBAS-1117> Add support for adding and removing function blocks over…

### PR DESCRIPTION
**Pull request by janm-dw**

### CODE REVIEW

File: shared/libraries/opcuatms/opcuatms/include/opcuatms/converters/dict_conversion_utils.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+#pragma once
+#include <opcuatms/opcuatms.h>
+#include <opcuashared/opcuavariant.h>
+#include <property_object_ptr.h>
+
+BEGIN_NAMESPACE_OPENDAQ_OPCUA_TMS
+
+class DictConversionUtils
```
Comment (janm-dw) [inline shared/libraries/opcuatms/opcuatms/include/opcuatms/converters/dict_conversion_utils.h:24]: I don't really like the name of this class, I am open for suggestions.

File: shared/libraries/opcuatms/tests/opcuatms_integration/test_tms_integration.cpp [hunk 2]
```diff
@@ -236,3 +237,84 @@ TEST_F(TmsIntegrationTest, GetDomainSignal)
...
+    TmsServer tmsServer(device);
+    tmsServer.start();
+
+    TmsClient tmsClient(device.getContext(), nullptr, OPC_URL, nullptr);
+    auto clientDevice = tmsClient.connect();
+
+    const auto clientFbTypes = clientDevice.getAvailableFunctionBlockTypes();
+
+    auto config = clientFbTypes.get("mock_fb_uid").createDefaultConfig();
+    config.addProperty(IntProperty("TestConfigInt", 0)); // I realy dont like this, but there is no way of trasfering default config property object over opcua ...
```
Comment (janm-dw) [inline shared/libraries/opcuatms/tests/opcuatms_integration/test_tms_integration.cpp:289]: Any suggestions about this? Maybe we could allow DictObject as config parameter.

File: shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_device_impl.cpp [hunk 3]
```diff
@@ -405,14 +407,82 @@ void TmsClientDeviceImpl::findAndCreateCustomComponents()
         this->components.push_back(val);
 }
 
-FunctionBlockPtr TmsClientDeviceImpl::onAddFunctionBlock(const StringPtr& /*typeId*/, const PropertyObjectPtr& /*config*/)
+DictPtr<IString, IFunctionBlockType> TmsClientDeviceImpl::onGetAvailableFunctionBlockTypes()
 {
-    throw OpcUaClientCallNotAvailableException();
+    const auto fbFolderNodeId = getNodeId("FB");
+    const auto methodNodeId = clientContext->getReferenceBrowser()->getChildNodeId(fbFolderNodeId, "GetAvailableFunctionBlockTypes");
```
Comment (JakaMohorkoDS) [inline shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_device_impl.cpp:413]: Could this just be a read-only array string node?

**Review by JakaMohorkoDS — APPROVED** [review]

File: shared/libraries/opcuatms/opcuatms/include/opcuatms/converters/dict_conversion_utils.h (hunk 1 above)
Comment (JakaMohorkoDS) [reply shared/libraries/opcuatms/opcuatms/include/opcuatms/converters/dict_conversion_utils.h:24]: If anything, this is PropertyObjectConversionUtils

File: shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_device_impl.cpp (hunk 3 above)
Comment (janm-dw) [reply shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_device_impl.cpp:413]: We have discussed this and decided to go with a folder node of available function block types.

**Review by JakaMohorkoDS — APPROVED** [review]

**Pull request merged**